### ISSUE TITLE
Gas/Electricity Current accumulative cost sensors not updating

### ISSUE THREAD

**Issue by jarrodstrachan96** [BottlecapDave/HomeAssistant-OctopusEnergy#358]
### Describe the bug

Gas/Electricity sensors are not updating, unless HomeAssistant is manually restarted. Current Demand sensor updates correctly every minute as detailed in docs. The changes in the graph you can see were prompted by a HA restart.
 
<img width="555" alt="image" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/81656574/d955751a-6acb-4b7d-aae4-b5cb0c7bc3de">

<img width="566" alt="image" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/81656574/ffa7890d-b579-4987-8ae4-d7f8fc27ce22">

### Reproduction steps

Installed via HACS. Using latest version which supports current accumulative cost sensors (via Octopus Home Mini, which I have).

### Expected behaviour

Sensors to update automatically, although I'm not sure what the update interval should be.

### Integration Version

8.0.0

### Home Assistant Version

2023.8.1

### Fresh Install?

Not specified

### Home Assistant Logs

No related logs

**Comment by BottlecapDave** [comment]
Hello and sorry that you're having issues. Are the non cost accumulative sensors updating? If you click "show more" do you get the same results? I only ask the latter as I've had a few times when the history view shows updated data vs the pop up history view.

I've had sporadic reports of various sensors not updating, but I'm struggling to track down the core issue.

**Comment by 78davidcross** [comment]
Could it be this:

[homeassistant.components.sensor.recorder] The unit of sensor.octopus_energy_gas_e6s20099782161_7577405905_current_rate (None) cannot be converted to the unit of previously compiled statistics (GBP/kWh). Generation of long term statistics will be suppressed unless the unit changes back to GBP/kWh or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this[0m

**Comment by BottlecapDave** [comment]
I don't believe that's it. I believe that appears when the integration is first starts because we don't have a valid value straight away. The screenshots show a value, it's just an old value, as if the update call isn't being made. I've been testing some changes to how the sensors are updated, but as I've never experienced this issue, it's difficult to see if it's improved things.

**Comment by BottlecapDave** [comment]
This should be fixed in the latest update, as there were some updates around how sensors update their data. Can anyone confirm if they are still seeing this?

**Comment by dannypetrov** [comment]
I have just switched from “Bright” (Hildebrandglow) to this, and I seem to observe the same behaviour. 

The only difference is that I have this in the log:

```
background
2023-09-03 08:10:46.783 WARNING (Recorder) [homeassistant.components.recorder.db_schema] State attributes for sensor.octopus_energy_electricity_21l3504461_1900017258302_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
2023-09-03 08:10:46.801 WARNING (Recorder) [homeassistant.components.recorder.db_schema] State attributes for sensor.octopus_energy_gas_e6s13655842061_565590303_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
2023-09-03 08:10:46.901 WARNING (Recorder) [homeassistant.components.recorder.db_schema] State attributes for sensor.octopus_energy_electricity_21l3504461_1900017258302_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
2023-09-03 08:10:46.907 WARNING (Recorder) [homeassistant.components.recorder.db_schema] State attributes for sensor.octopus_energy_gas_e6s13655842061_565590303_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
```

**Comment by dannypetrov** [comment]
Hey Dave, thanks for your response. Sadly, I haven't got any time to troubleshoot this any further myself, but I just ran a quick check and indeed the reported metrics I see in [here](https://api.octopus.energy/v1/gas-meter-points/xxxxxxxxx/meters/xxxxxxxxx/consumption/) seem a bit off (since yesterday / EOD yesterday). 

Is this something common for Octopus based on your experience or is it something with my account? I can see all my data accurately on their page' dashboard - it just seems that it's not getting exposed through the API for whatever reason...Apologies for the noise anyway, I should have done some basic checks before I chipped in.

**Comment by BottlecapDave** [comment]
@dannypetrov Some users have a delay in when their data is reported from their smart meter to the OE servers. Any data that is displayed in the OE dashboard is accessible via the integration, however the sensors only look at the previous 24 hours so if it takes longer than 24 hours to receive the data, the sensors will not pick it up. I have it on my list to provide a potential solution for this scenario.

**Comment by BottlecapDave** [comment]
I've added the ability to set the offset for the previous consumption sensors in scenarios where OE are delayed in obtaining your information. This can be found in [v8.3.0](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/v8.3.0), and the docs can be found at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/setup_account.md#previous-consumption-days-offset

I'm also working on a separate service for importing data from adhoc periods (like in the scenario of a day of data being missing), so I'll keep this issue open until that has been added. 

I'm going to close this issue, as it now appears to be a duplicate of https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/312.
